walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,79 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Set<String> attendees = new HashSet<String>(request.getAttendees());
+    List<TimeRange> initalRanges = new ArrayList<TimeRange>();
+    List<TimeRange> results = new ArrayList<TimeRange>();
+    final int minMeetingTime = 15;
+    
+    for(int start = 0; start <= TimeRange.END_OF_DAY; start += minMeetingTime) {
+        if(start + minMeetingTime > TimeRange.END_OF_DAY) {
+            initalRanges.add(TimeRange.fromStartEnd(start, TimeRange.END_OF_DAY, false));
+        } else {
+            initalRanges.add(TimeRange.fromStartDuration(start, minMeetingTime));
+        }
+    }
+
+    for(Event currEvent : events) {
+        Set<String> currAttendees = currEvent.getAttendees();
+        for(String attendee : currAttendees) {
+            if(attendees.contains(attendee)) {
+                blockTimeRange(initalRanges, currEvent);
+                break;
+            }
+        }
+    }
+
+    TimeRange tempRange = new TimeRange();
+    int duration = (int) request.getDuration();
```

Comment:
why does this need a cast?

Reply:
Originally the request file had duration as a long but it is an int in other files so I just changed to request file so it will be an int there now too

Reply:
After doing this it broke a lot regarding the tests but I could not figure out why it was requiring a long since the tests were passing in ints anyways
 ex. CantAddOptionalAttendeeWhoIsAlsoMandatory(com.google.sps.MeetingRequestTest): com.google.sps.MeetingRequest.<init>(Ljava/util/Collection;J)V